### PR TITLE
Button: Stopping event propagation on click

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This PR adds `ev.stopPropagation()` in the `onClick` callbacks of the experimental `Button`.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8715)

### CODE REVIEW

**Comment by khmakoto** [comment]
@msft-github-bot require review from @JasonGore

**Comment by khmakoto** [comment]
@msft-github-bot make sure @JasonGore or @dzearing review this before merging

**Comment by cliffkoh** [comment]
@msft-github-bot require at least three point one four one six approvers

**Comment by JasonGore** [comment]
@msft-github-bot require at least six point two eight three one approvals

**Comment by JasonGore** [comment]
@msft-github-bot I want not one but two pies

File: packages/experiments/src/components/Button/Button.view.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ export const ButtonView: IButtonComponent['view'] = props => {
   const _onClick = (ev: React.MouseEvent<HTMLElement>) => {
     if (!disabled && onClick) {
       onClick(ev);
+      ev.stopPropagation();
```
Comment (JasonGore) [inline packages/experiments/src/components/Button/Button.view.tsx]: Is this consistent with existing buttons in OUFR? Seems a bit different.

Also linse 27-29 are weird. They're basically no-op.

Comment (khmakoto) [reply packages/experiments/src/components/Button/Button.view.tsx]: I think lines 27-29 were something that was left after I split `MenuButton` into its own variant. I'll remove them. Also, I think you're right, in the current button `ev.stopPropagation()` only occurs `onMenuClick`, so I'll also remove that from here.

**Comment by apneer** [comment]
Component Perf Results:
  <table>
  <tr>
    <th>Scenario</th>
    <th>Current Avg Total (ms)</th>
    <th>New Avg Total (ms)</th>
    <th>Current Avg Per Item (ms)</th>
    <th>New Avg Per Item (ms)</th>
  </tr><tr>
            <td>PrimaryButton</td>
            <td>97.970</td>
            <td>100.919</td>
            <td>0.980</td>
            <td>1.009</td>
           </tr>  
<tr>
            <td>BaseButton</td>
            <td>33.371</td>
            <td>36.237</td>
            <td>0.334</td>
            <td>0.362</td>
           </tr>  
<tr>
            <td>NewButton</td>
            <td>106.754</td>
            <td>106.070</td>
            <td>1.067</td>
            <td>1.061</td>
           </tr>  
<tr>
            <td>button</td>
            <td>5.477</td>
            <td>4.986</td>
            <td>0.055</td>
            <td>0.050</td>
           </tr>  
<tr>
            <td>DetailsRows without styles</td>
            <td>184.762</td>
            <td>180.968</td>
            <td>1.848</td>
            <td>1.810</td>
           </tr>  
<tr>
            <td>DetailsRows</td>
            <td>195.517</td>
            <td>200.378</td>
            <td>1.955</td>
            <td>2.004</td>
           </tr>  
<tr>
            <td>Toggles</td>
            <td>49.864</td>
            <td>49.941</td>
            <td>0.499</td>
            <td>0.499</td>
           </tr>  
<tr>
            <td>NewToggle</td>
            <td>68.117</td>
            <td>68.356</td>
            <td>0.681</td>
            <td>0.684</td>
           </tr>  </table><div id="perfComment9423"/>

File: packages/experiments/src/components/Button/MenuButton/MenuButton.state.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export class MenuButtonState extends BaseState<IMenuButtonProps, IMenuButtonView
     if (!disabled) {
       if (onClick) {
         onClick(ev);
+        ev.stopPropagation();
```
Comment (dzearing) [inline packages/experiments/src/components/Button/MenuButton/MenuButton.state.ts]: This seems wrong.

Why are you stopping propogation only if onClick is passed in?

Comment (dzearing) [reply packages/experiments/src/components/Button/MenuButton/MenuButton.state.ts]: I think stopping propogation might be OK if we are doing work for the click and don't expect other clicks to be handled at a higher layer. It just seems wrong to do it inside of the if check.

Comment (khmakoto) [reply packages/experiments/src/components/Button/MenuButton/MenuButton.state.ts]: We do want it to stop propagation only when the `MenuButton` is not `disabled` right?

**Pull request closed without merging**